android-agent/src/main/kotlin/io/opentelemetry/android/agent/AndroidAgent.kt

```diff
@@ -0,0 +1,167 @@
+import io.opentelemetry.android.OpenTelemetryRum
+import io.opentelemetry.android.OpenTelemetryRumBuilder
+import io.opentelemetry.android.agent.endpoint.EndpointConfig
+import io.opentelemetry.android.agent.session.SessionIdTimeoutHandler
+import io.opentelemetry.android.agent.session.SessionManager
+import io.opentelemetry.android.config.OtelRumConfig
+import io.opentelemetry.android.features.diskbuffering.DiskBufferingConfiguration
+import io.opentelemetry.android.instrumentation.AndroidInstrumentationLoader
+import io.opentelemetry.android.instrumentation.activity.ActivityLifecycleInstrumentation
+import io.opentelemetry.android.instrumentation.anr.AnrInstrumentation
+import io.opentelemetry.android.instrumentation.common.ScreenNameExtractor
+import io.opentelemetry.android.instrumentation.crash.CrashDetails
+import io.opentelemetry.android.instrumentation.crash.CrashReporterInstrumentation
+import io.opentelemetry.android.instrumentation.fragment.FragmentLifecycleInstrumentation
+import io.opentelemetry.android.instrumentation.network.NetworkChangeInstrumentation
+import io.opentelemetry.android.instrumentation.slowrendering.SlowRenderingInstrumentation
+import io.opentelemetry.android.internal.services.ServiceManager
+import io.opentelemetry.android.internal.services.network.data.CurrentNetwork
+import io.opentelemetry.api.trace.Tracer
+import io.opentelemetry.exporter.otlp.http.logs.OtlpHttpLogRecordExporter
+import io.opentelemetry.exporter.otlp.http.trace.OtlpHttpSpanExporter
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor
+import io.opentelemetry.sdk.common.Clock
+import java.time.Duration
+
+object AndroidAgent {
+    private val activityLifecycleInstrumentation by lazy {
+        AndroidInstrumentationLoader.getInstrumentation(
+            ActivityLifecycleInstrumentation::class.java,
+        )!!
+    }
+    private val fragmentLifecycleInstrumentation by lazy {
+        AndroidInstrumentationLoader.getInstrumentation(
+            FragmentLifecycleInstrumentation::class.java,
+        )!!
+    }
+    private val anrInstrumentation by lazy {
+        AndroidInstrumentationLoader.getInstrumentation(
+            AnrInstrumentation::class.java,
+        )!!
+    }
+    private val crashReporterInstrumentation by lazy {
+        AndroidInstrumentationLoader.getInstrumentation(
+            CrashReporterInstrumentation::class.java,
+        )!!
+    }
+    private val networkChangeInstrumentation by lazy {
+        AndroidInstrumentationLoader.getInstrumentation(NetworkChangeInstrumentation::class.java)!!
+    }
+    private val slowRenderingInstrumentation by lazy {
+        AndroidInstrumentationLoader.getInstrumentation(SlowRenderingInstrumentation::class.java)!!
+    }
+
+    fun createRumBuilder(
+        application: Application,
+        otelRumConfig: OtelRumConfig = OtelRumConfig(),
+        endpointConfig: EndpointConfig = EndpointConfig.getDefault("http://localhost"),
```

Comment:
I like the idea of having a simple and direct configuration approach, but we should still make it easy for users to configure separate endpoints for logs and spans. We can't assume that they're the same.

Reply:
Got it, yeah it makes sense. Right now the [EndpointConfig](https://github.com/open-telemetry/opentelemetry-android/pull/624/files#diff-9c2d3bac896bbf1cfba28553fec9c4b8ff7c8016e668013381080ec04f28c287) interface provides a separate getter for each, although it's not the most user-friendly approach having to implement it just to provide different URLs. Do you have any preferences of what could be a nice way to address this? The one thing I was thinking right now was to remove the `EndpointConfig` param and instead add string params for each url, and maybe a "defaultUrl" one that gets overridden if there's a value provided for a specific signal url.

Reply:
Separate for each and fallback to the default sounds good. There aren't so many signals that it would be onerous to specify the URL twice on init if they have a custom one